src/iter/collect/mod.rs

```diff
@@ -64,6 +66,24 @@ where
     });
 }
 
+/// Collects the iterator into a linked list of vectors.
+///
+/// This is called by `ParallelIterator::collect_vec_list`.
+pub(super) fn collect_vec_list<I>(pi: I) -> LinkedList<Vec<I::Item>>
+where
+    I: ParallelIterator,
+{
+    match pi.opt_len() {
+        Some(len) => {
+            // Pseudo-specialization. See impl of ParallelExtend for Vec for more details.
+            let mut v = Vec::new();
+            super::collect::special_extend(pi, len, &mut v);
+            LinkedList::from([v])
```

Comment:
Wasn't sure if I should re-use `pi.collect::<Vec<_>>()` here or not; theoretically that'd have an unnecessary second check of opt_len() but that would get optimized in almost all cases.